lxd/device/nic_routed.go

```diff
@@ -205,6 +207,42 @@ func (d *nicRouted) validateEnvironment() error {
 	return nil
 }
 
+func (d *nicRouted) checkIPAvailability(parent string) error {
+	var addresses []net.IP
+	ipv4AddrStr, ok := d.config["ipv4.address"]
+	if ok {
+		ipv4Addrs := util.SplitNTrimSpace(ipv4AddrStr, ",", -1, true)
+		for _, addr := range ipv4Addrs {
+			addresses = append(addresses, net.ParseIP(addr))
+		}
+	}
+
+	ipv6AddrStr, ok := d.config["ipv6.address"]
+	if ok {
+		ipv6Addrs := util.SplitNTrimSpace(ipv6AddrStr, ",", -1, true)
+		for _, addr := range ipv6Addrs {
+			addresses = append(addresses, net.ParseIP(addr))
+		}
+	}
+
+	errs := make(chan error, len(addresses))
+	for _, address := range addresses {
+		go func(address net.IP) {
+			ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+			defer cancel()
+			errs <- isIPAvailable(ctx, address, parent)
+		}(address)
+	}
+
+	for range addresses {
+		if err := <-errs; err != nil {
```

Comment:
In general we're staying away from this short syntax in LXD code.
We don't like the define + check syntax as it's hard to read and can lead to variable scoping weirdness.

So should we should change that to a standard two line set + check.

Reply:
Ah yes I remember - fixed now.